components/user-self-registration/org.wso2.carbon.identity.user.registration.mgt/src/main/java/org/wso2/carbon/identity/user/registration/mgt/dao/RegistrationFlowDAOImpl.java

```
@@ -0,0 +1,279 @@
+                                              .height(resultSet.getDouble(HEIGHT))
+                                              .width(resultSet.getDouble(WIDTH))
+                                              .build();
+
+                                      resolvePageContent(stepDTO, resultSet.getBinaryStream(PAGE_CONTENT), tenantId);
+                                      return stepDTO;
+                                  })), preparedStatement -> preparedStatement.setInt(1, tenantId));
+
+            RegistrationFlowDTO registrationFlowDTO = new RegistrationFlowDTO();
+            if (steps.isEmpty()) {
+                LOG.debug("No steps are found in the default flow of tenant " + tenantId);
+                return registrationFlowDTO;
+            }
+            registrationFlowDTO.getSteps().addAll(steps);
+            return registrationFlowDTO;
+        } catch (DataAccessException e) {
+            throw handleServerException(Constants.ErrorMessages.ERROR_CODE_GET_DEFAULT_FLOW, e, tenantId);
+        }
+    }
+
+    private void resolvePageContent(StepDTO stepDTO, InputStream pageContent, int tenantId)
+            throws DataAccessException, RegistrationServerException {
+
+        try (ObjectInputStream ois = new ObjectInputStream(pageContent)) {
+            Object obj = ois.readObject();
+            if (VIEW.equals(stepDTO.getType()) && obj instanceof List<?>) {
+                List<?> tempList = (List<?>) obj;
+                if (!tempList.isEmpty() && tempList.get(0) instanceof ComponentDTO) {
+                    List<ComponentDTO> components = tempList.stream()
+                            .map(ComponentDTO.class::cast)
+                            .collect(Collectors.toList());
+                    stepDTO.setData(new DataDTO.Builder().components(components).build());
+                } else {
+                    LOG.error("Deserialized list does not contain component objects.");
```

should we throw a server error here